plugins/modules/ndo_mcp_global_policy.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Samita Bhattacharjee (@samiib) <samitab@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_mcp_global_policy
+short_description: Manage the MCP Global Policy in a Fabric Policy Template on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage the MisCabling Protocol (MCP) Global Policy in a Fabric Policy Template on Cisco Nexus Dashboard Orchestrator (NDO).
+- There can only be a single MCP Global Policy in a Fabric Policy Template.
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Samita Bhattacharjee (@samiib)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a Fabric Policy template.
+    type: str
+    aliases: [ fabric_template ]
+    required: true
+  name:
+    description:
+    - The name of the MCP Global Policy.
+    type: str
+    aliases: [ mcp_global_policy ]
+  uuid:
+    description:
+    - The UUID of the MCP Global Policy.
+    type: str
+    aliases: [ mcp_global_policy_uuid ]
+  description:
+    description:
+    - The description of the MCP Global Policy.
+    - Providing an empty string will remove the O(description="") from the MCP Global Policy.
+    type: str
+  admin_state:
+    description:
+    - The administrative state of the MCP Global Policy.
+    - Defaults to C(enabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  key:
+    description:
+    - The key of the MCP Global Policy.
+    type: str
+  per_vlan:
+    description:
+    - Enable or disable MCP packets being sent to each End Point Group (EPG).
```

Comment:
Noot sure the description fully matches the documentation explanation:

This enables MCP to send packets on a per-EPG basis. If this option is disabled, the packets will only be sent on untagged EPGs which allows detecting loops in the native VLAN only.

Reply:
The enabled description sounds like it matches to me. However, I will add the information when the option is disabled.
Would it be okay to share the link to the documentation?

Reply:
Ok perhaps it is my perception that "to each EPG" vs "a per-EPG basis" is slightly different. So never mind if that is the same, would add the disabled text.

What do you mean with link to the documentation? In the description a link to https://www.cisco.com/c/en/us/td/docs/dcn/ndo/4x/configuration/cisco-nexus-dashboard-orchestrator-configuration-guide-aci-421/ndo-configuration-aci-fabric-management-42x.html? Or as general in the see also part?

Reply:
I was just curious to see the documentation you referred to, not to add it to the module.